lib/order.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+
+    [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status) ? ( @fulfillment_status = fulfillment_status ) : ( raise ArgumentError.new("Invalid fulfillment status.") )
+  end
+
+  def total
+    total_cost = 0
+
+    @products.each_value { |price| total_cost += price }
+
+    total_cost *= 1.075
+    return total_cost.round(2)
+  end
+
+  def add_product(product_name, price)
+    @products.keys.include?(product_name) ? ( raise ArgumentError.new("Product with that name has already been added to the order.") ) : ( @products[product_name] = price )
+  end
+
+  def remove_product(product_name)
+    @products.keys.include?(product_name) ? ( @products.delete(product_name) ) : ( raise ArgumentError.new("Product with that name was not found.") )
```

Comment:
I'm not sure the parentheses are necessary here. If they are, I would consider making this a traditional conditional rather than a ternary to increase readability. You've packed a lot in one line. Another idea is to use a guard clause.

```
raise ArgumentError.new("Product with that name was not found.") if !@products.keys.include?(product_name)
@products.delete(product_name)
```